From f747a38746cbf0fa4279e773835b7d872c0d313c Mon Sep 17 00:00:00 2001
From: Guillem Jover <guillem@debian.org>
Date: Sun, 3 Mar 2024 18:42:34 +0100
Subject: [PATCH] Dpkg::Vendor::Debian: Make it possible to disable
 qa=-bug-implicit-func

We do not need to forcibly enable this feature if the user explicitly
specified it.

Closes: #1065371
---
 scripts/Dpkg/Vendor/Debian.pm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/Dpkg/Vendor/Debian.pm b/scripts/Dpkg/Vendor/Debian.pm
index fcf5b1e2a..ad727d2cf 100644
--- a/scripts/Dpkg/Vendor/Debian.pm
+++ b/scripts/Dpkg/Vendor/Debian.pm
@@ -299,8 +299,8 @@ sub set_build_features {
         $use_feature{abi}{lfs} = 1 if $libc eq 'gnu';
 
         # Require -Werror=implicit-function-declaration, to avoid linking
-        # against the wrong symbol.
-        $use_feature{qa}{'bug-implicit-func'} = 1;
+        # against the wrong symbol, unless it has been set explicitly.
+        $use_feature{qa}{'bug-implicit-func'} //= 1;
     }
 
     # XXX: Handle lfs alias from future abi feature area.
-- 
2.43.0

